core/src/test/java/com/linecorp/armeria/client/AdditionalAuthorityTest.java

```diff
@@ -0,0 +1,184 @@
+
+            sb.virtualHost("bar")
+              .service("/", (ctx, req) -> HttpResponse.of("bar/" + req.authority()));
+        }
+    };
+
+    @RegisterExtension
+    static ServerExtension serverB = new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) {
+            sb.service("/", (ctx, req) -> HttpResponse.of("serverB/" + req.authority()));
+        }
+    };
+
+    private static WebClient client;
+    private static int serverAPort;
+
+    @BeforeAll
+    static void beforeAll() {
+        final ClientFactory clientFactory =
+                ClientFactory.builder()
+                             .addressResolverGroupFactory(
+                                     eventLoop -> MockAddressResolverGroup.localhost())
+                             .build();
+
+        client = WebClient.builder()
+                          .factory(clientFactory)
+                          .build();
+        serverAPort = serverA.httpPort();
+    }
+
+    @AfterAll
+    static void afterAll() {
+        client.options().factory().closeAsync();
+    }
+
+    @Test
+    void additionalAuthorityHasHighestPrecedent() {
+        try (SafeCloseable ignored = Clients.withContextCustomizer(
+                ctx -> ctx.addAdditionalRequestHeader(HttpHeaderNames.AUTHORITY,
+                                                      "bar:" + serverAPort))) {
+
+            assertThat(client.get("http://foo:" + serverAPort).aggregate().join().contentUtf8())
+                    .isEqualTo("bar/bar:" + serverAPort);
+        }
+    }
+
+    @Test
+    void requestHeader() {
+        final HttpRequest request = HttpRequest.of(RequestHeaders.builder(HttpMethod.GET, "/")
+                                                                 .scheme("http")
+                                                                 .authority("bar:" + serverAPort)
+                                                                 .build());
+        assertThat(client.execute(request).aggregate().join().contentUtf8())
+                .isEqualTo("bar/bar:" + serverAPort);
+    }
+
+    @Test
+    void noAuthority() {
+        final HttpRequest request = HttpRequest.of(RequestHeaders.of(HttpMethod.GET, "/"));
+        assertThatThrownBy(() -> client.execute(request).aggregate().join())
+                .isInstanceOf(CompletionException.class)
+                .hasCauseInstanceOf(IllegalArgumentException.class)
+                .hasMessageContaining("Scheme and authority must be specified in");
+    }
+
+    @Test
+    void absolutePath() {
+        final HttpRequest request =
+                HttpRequest.of(RequestHeaders.of(HttpMethod.GET, "http://foo:" + serverAPort));
+        assertThat(client.execute(request).aggregate().join().contentUtf8()).isEqualTo(
+                "foo/foo:" + serverAPort);
+    }
+
+    @Test
+    void shouldIgnoreInvalidAdditionalAuthority() {
```

Comment:
Should we fail the request instead of ignoring an invalid value?

Reply:
I followed the original policy but it makes sense to fail the request. Let me address the comment.